src/lib.rs

```diff
@@ -61,20 +63,23 @@ pub struct Metadata {
     /// The crate name
     pub name: Cow<'static, str>,
     /// The list of authors of the crate
-    pub authors: Cow<'static, str>,
+    pub authors: MaybeString,
     /// The URL of the crate's website
-    pub homepage: Cow<'static, str>,
+    pub homepage: MaybeString,
+    /// The support information
+    pub support: MaybeString,
 }
 
-/// Initialize [`Metadata`]
+/// Initialize [`Metadata`].
 #[macro_export]
 macro_rules! metadata {
     () => {{
         $crate::Metadata {
             version: env!("CARGO_PKG_VERSION").into(),
             name: env!("CARGO_PKG_NAME").into(),
-            authors: env!("CARGO_PKG_AUTHORS").replace(":", ", ").into(),
-            homepage: env!("CARGO_PKG_HOMEPAGE").into(),
+            authors: Some(env!("CARGO_PKG_AUTHORS").replace(":", ", ").into()),
+            homepage: None,
```

Comment:
The all `Option` suggest has a trouble that what if we meet `Some("")`?

Like for `CARGO_PKG_HOMEPAGE`, it's common that a crate doesn't configure this field.